Creates types incompatible with typescript 3.x

See this repo: https://github.com/Narretz/mongoose-tsgen-issues/tree/typescript (make sure to check out the **typescript** branch)

I don't think it's documented explicitly which typescript versions are supported, but the package.json itself requires ^3.9.7. However, if your own code is requiring typescript 3.9.7, it won't work, because te generated type will throw a bunch of errors:

```
src/interfaces/mongoose.gen.ts:268:35 - error TS2344: Type 'Unarray<ResultType>' does not satisfy the constraint 'Document<any, any>'.
  Type '{}' is missing the following properties from type 'Document<any, any>': $getAllSubdocs, $ignore, $isDefault, $isDeleted, and 48 more.

268         ? Array<PopulatedDocument<Unarray<ResultType>, T>>
                                      ~~~~~~~~~~~~~~~~~~~

src/interfaces/mongoose.gen.ts:270:29 - error TS2344: Type 'Unarray<ResultType>' does not satisfy the constraint 'Document<any, any>'.
  Type 'unknown' is not assignable to type 'Document<any, any>'.

270         ? PopulatedDocument<Unarray<ResultType>, T>
                                ~~~~~~~~~~~~~~~~~~~

src/interfaces/mongoose.gen.ts:281:35 - error TS2344: Type 'Unarray<ResultType>' does not satisfy the constraint 'Document<any, any>'.

281         ? Array<PopulatedDocument<Unarray<ResultType>, T>>
                                      ~~~~~~~~~~~~~~~~~~~

src/interfaces/mongoose.gen.ts:283:29 - error TS2344: Type 'Unarray<ResultType>' does not satisfy the constraint 'Document<any, any>'.

283         ? PopulatedDocument<Unarray<ResultType>, T>
                                ~~~~~~~~~~~~~~~~~~~
```

--- a/README.md
+++ b/README.md
@@ -349,4 +349,4 @@ export type UserDocument = mongoose.Document<mongoose.Types.ObjectId, UserQuerie
 
 ## Development
 
-- [ ] Cut down node_modules by using peer dependencies (i.e. mongoose) and stripping oclif.
+- [ ] Remove oclif - it adds a lot of unnecessary dependencies

--- a/package.json
+++ b/package.json
@@ -21,13 +21,16 @@
     "glob": "^7.1.6",
     "lodash": "^4.17.20",
     "mkdirp": "^1.0.4",
-    "mongoose": "^5.12.10",
     "prettier": "^2.1.2",
     "strip-json-comments": "^3.1.1",
     "ts-morph": "^8.1.2",
     "ts-node": "^9.0.0",
     "tsconfig-paths": "^3.9.0",
     "tslib": "^1.13.0"
+  },
+  "peerDependencies": {
+    "mongoose": ">= 5.11.19",
+    "typescript": ">= 4"
   },
   "devDependencies": {
     "@oclif/dev-cli": "^1.22.2",
@@ -44,9 +47,10 @@
     "husky": "^4.3.0",
     "jest": "^26.4.2",
     "lint-staged": "^10.5.1",
+    "mongoose": "^5.12.10",
     "rimraf": "^3.0.2",
     "ts-jest": "^26.4.1",
-    "typescript": "^3.9.7"
+    "typescript": "^4.1.3"
   },
   "engines": {
     "node": ">=9.0.0"

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "mongoose-tsgen",
   "description": "A Typescript interface generator for Mongoose that works out of the box.",
-  "version": "8.3.4",
+  "version": "8.3.5",
   "author": "Francesco Virga @francescov1",
   "bin": {
     "mtgen": "./bin/run"

